### DISCUSSION TITLE
ModuleNotFoundError: No module named 'XY'

### DISCUSSION THREAD

**Discussion by DerErzengel** [pythonnet/pythonnet#2512] [Q&A]
I want to work with a tool called Archi Steam Farm, if that tells anyone anything. This is written in C#, which is why I came up with the idea of being able to call the functions via pythonnet in python.  

My python code is as follows:  (sorry, for some reason the code blocks are not formatted...)

```python
import clr
import sys

clr.AddReference(r"C:\Users\XXXX\bin\Debug\net8.0\MyAwesomePlugin.dll")

from MyAwesomePlugin import MyAwesomePlugin`
```

and the C# code:

```csharp
namespace MyAwesomePlugin;

#pragma warning disable CA1812 // ASF uses this class during runtime
[UsedImplicitly]
public class MyAwesomePlugin : IGitHubPluginUpdates {
    public string Name => nameof(MyAwesomePlugin);
    public string RepositoryName => "JustArchiNET/ASF-PluginTemplate";
    public Version Version => typeof(MyAwesomePlugin).Assembly.GetName().Version ?? throw new InvalidOperationException(nameof(Version));

    public Task OnLoaded() {
        ASF.ArchiLogger.LogGenericInfo($"Hello {Name}!");
        ASF.ArchiLogger.LogGenericInfo($"Es funktioniert JAJAJAJAJA!");
        return Task.CompletedTask;
    }

    public static async Task GetBotInventory(string bot, string SteamID) {
        
        
        if (true) {
            var inventory = Bot.GetBot("bot16").ArchiHandler.GetMyInventoryAsync(730, 2);


            if (inventory != null) {
                ASF.ArchiLogger.LogGenericInfo($"Inventar für Bot 'bot16' erfolgreich abgerufen. Anzahl Items:");
                string jsonString = JsonSerializer.Serialize(inventory, new JsonSerializerOptions { WriteIndented = true });
                Console.WriteLine(jsonString);
            } else {
                ASF.ArchiLogger.LogGenericWarning("Konnte das Inventar nicht abrufen.");
            }
        }
    }
```

Unfortunately I always get this error message: 

```python
from MyAwesomePlugin import GetBotInventory
```

```
ModuleNotFoundError: No module named 'MyAwesomePlugin'`
```

I have seen this error several times on the Internet, but have not yet been able to find a fix that has resolved it for me. Does anyone have an idea what the problem could be?

Thanks for your time and help! 🙏

**Comment by filmor** [comment]
`import clr` defaults to .NET Framework on Windows (and Mono on non-Windows), but your library is compiled against .NET (Core). You have to load the correct runtime explicitly:

```python
import pythonnet
pythonnet.load("coreclr")

import clr
```

https://pythonnet.github.io/pythonnet/python.html#calling-pythonnet-load

**Reply by DerErzengel** [reply]
Thank you very much for the reply and quick help!

Unfortunately, I still have the problem. I am also not very technically fit in programming, I hope that I am not just incompetent. This is my code:

`import pythonnet
import sys

pythonnet.load("coreclr")

import clr

assembly_path2 = r"C:\Users\Gabriel\Documents\OneDrive\Projekte\PluginwithArchi\ASF-PluginTemplate\MyAwesomePlugin\bin\Debug\net8.0"

sys.path.append(assembly_path2)

clr.AddReference("MyAwesomePlugin")

from MyAwesomePlugin import MyAwesomePlugin`

Surprisingly, the import works if I follow the instructions exactly (with the help of sys.path.append): 

https://medium.com/@mhmdjmala51/using-c-in-python-is-not-difficult-anymore-dbe5c3251431

However, it doesn't work with my own DDL. I have installed .Net 8.0.403

Since I hardly know anything about Net, it may very well be that there is some kind of incompatibility here that I don't notice. 
I use a template C# file from this Archi Steam Farm project:

https://github.com/JustArchiNET/ASF-PluginTemplate
